spectator/stateless_meters.h

```diff
@@ -6,14 +6,43 @@
 namespace spectator {
 
 namespace detail {
+
 inline std::string as_string(std::string_view v) { 
     return {v.data(), v.size()}; 
 }
+
+inline bool is_valid_character(char c) {
+  return (c >= 'a' && c <= 'z') ||
+         (c >= 'A' && c <= 'Z') ||
+         (c >= '0' && c <= '9') ||
+         c == '-' ||
+         c == '.' ||
+         c == '_' ||
+         c == '~' ||
+         c == '^';
+}
+
+inline std::string replace_invalid_characters(const std::string& input) {
```

Comment:
This will create a new string for each key and value processed, a more efficient implementation is to check whether that is needed, and only do it if required. See: https://github.com/Netflix/spectator/blob/main/spectator-api/src/main/java/com/netflix/spectator/impl/AsciiSet.java#L208

Reply:
Does the original string value need to be maintained? Since `std::string` in C++ is mutable you could just update it in place with something like `std::transform`.

Reply:
Yeah, I talked about that with Daniel yesterday, still working through those changes.